optimum/habana/checkpoint_utils.py

```diff
@@ -53,13 +54,27 @@ def get_repo_root(model_name_or_path, local_rank=-1, token=None):
 def get_checkpoint_files(model_name_or_path, local_rank, token=None):
     """
     Gets the list of files for the specified model checkpoint.
+    Copied from https://github.com/huggingface/transformers/blob/abbffc4525566a48a9733639797c812301218b83/src/transformers/modeling_utils.py#L414
```

Comment:
@vivekgoe yes it's ready to merge

@tjs-intel I see you opened a PR https://github.com/HabanaAI/optimum-habana-fork/pull/25. Please go ahead and implement additional changes on top of this to make it work for your model

Reply:
I can make my changes based on this branch, but the maintainers should know that this PR could cause a regression with models that don't use safetensors and only have a single checkpoint. I think this type of model probably exists because there are code paths for it in that `transformers` code snippet I linked.

Reply:
I added support for single weight checkpoints in #25 and changed the base to this branch